packages/shared_preferences/shared_preferences_web/lib/shared_preferences_web.dart

```diff
@@ -60,9 +64,13 @@ class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
     return true;
   }
 
-  Iterable<String> _getStoredFlutterKeys(String prefix) {
-    return html.window.localStorage.keys
-        .where((String key) => key.startsWith(prefix));
+  Iterable<String> _getStoredFlutterKeys(
+    String prefix, {
+    Set<String>? allowList,
+  }) {
+    return html.window.localStorage.keys.where((String key) =>
+        key.startsWith(prefix) &&
+        (allowList == null || allowList.contains(key)));
```

Comment:
```suggestion
        allowList?.contains(key) ?? true);
```

(Shorter but not sure if more readable :))

Also (and this is probably a worthless micro-optimization): should the "contains" operation happen before the "startsWith", so the where can bail out fast?